docs/addons/efs-csi-driver.md

```diff
@@ -0,0 +1,55 @@
+# EFS CSI Driver Amazon EKS Add-on
+
+The `EFS CSI Driver Amazon EKS Add-on` provides a CSI interface that allows Kubernetes clusters running on AWS to manage the lifecycle of Amazon EFS volumes for persistent storage.
+EFS CSI driver supports both dynamic and static provisioning of storage
+
+A couple of things to note:
+
+- Driver is not compatible with Windows-based container images
+- The number of replicas to be deployed must be less or equal to the number of nodes in the cluster
+
+For more information on the driver, please review the [user guide](https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html).
+
+## Prerequisites
+
+- The EFS file system itself must be created in AWS separately as the driver uses the EFS for storage, but it does not create it.
+
+## Usage
+
+```typescript
+import { App } from '@aws-cdk/core';
+import * as ssp from '@aws-quickstart/ssp-amazon-eks';
+
+const app = new App();
+
+ssp.EksBlueprint.builder()
+    .addOns(new ssp.EfsCsiDriverAddOn() )
+    .build(app, 'my-cluster');
+```
+
+## Configuration Options
+
+- `version`: Version of the EFS CSI Driver add-on to be installed. Version 2.2.3 will be installed by default if a value is not provided
+- `replicaCount`: Number of replicas to be deployed. If not provided, two replicas will be deployed. Note that the number of replicas
```

Comment:
can we assess minSize on the node group to determine the low boundary for the replicas? What if somebody rolls out a 1 node cluster as an experiment?

Reply:
Fair point. but should it check min size or desired size?

Reply:
@shapirov103 How can I get the minsize (or desired size) from within the add-on? Also, what if they are using Karpenter ? As far as I know Karpenter does not use node groups and just limits on [CPUs or Memory](https://karpenter.sh/v0.7.2/tasks/set-resource-limits/)

Reply:
@camba1 fair point :).  I see that once nodegroup is created you cannot really get the minSize parameter. We can get all the config options from the blueprint/add-on configuration but it is not going to be worth the effort. The current approach is simple and straightforward.